test/tests/parser/param_duplicate_check/tests

```diff
@@ -0,0 +1,10 @@
+[Tests]
+  design = 'Parser.md'
+  issues = '#25994'
+  [dup_check]
```

Comment:
What new capability was added to require this test? OR is this just adding coverage that didn't exist before?

Reply:
This is more like a patch to the existing duplication check. Joshua found the duplication check in parser doesn't work properly when multiple input files are merged together(only the last input file get checked). I changed the sequence of duplication and file merge to fix it. Probably should start a new PR on this, but it's very simple so I throw it here.